<h1>Iris Flowers Classification ML Project </h1>

<h2>Importing pandas</h2>

In [1]:
import pandas as pd

In [2]:
data=pd.read_csv("iris_data")

In [3]:
data.columns=['sepal_length','sepal_width','petal_length','petal_width','class']

In [4]:
data.to_csv("iris_data.csv",index=None)
data

sepal_length  sepal_width  petal_length  petal_width           class
0             4.9          3.0           1.4          0.2     Iris-setosa
1             4.7          3.2           1.3          0.2     Iris-setosa
2             4.6          3.1           1.5          0.2     Iris-setosa
3             5.0          3.6           1.4          0.2     Iris-setosa
4             5.4          3.9           1.7          0.4     Iris-setosa
..            ...          ...           ...          ...             ...
144           6.7          3.0           5.2          2.3  Iris-virginica
145           6.3          2.5           5.0          1.9  Iris-virginica
146           6.5          3.0           5.2          2.0  Iris-virginica
147           6.2          3.4           5.4          2.3  Iris-virginica
148           5.9          3.0           5.1          1.8  Iris-virginica

[149 rows x 5 columns]

Converting Categorical values to numeric values by label encoder

In [5]:
from sklearn.preprocessing import LabelEncoder

In [6]:
label_encode=LabelEncoder()

In [7]:
data['class_type']=label_encode.fit_transform(data['class'])

In [8]:
data

sepal_length  sepal_width  petal_length  petal_width           class  \
0             4.9          3.0           1.4          0.2     Iris-setosa   
1             4.7          3.2           1.3          0.2     Iris-setosa   
2             4.6          3.1           1.5          0.2     Iris-setosa   
3             5.0          3.6           1.4          0.2     Iris-setosa   
4             5.4          3.9           1.7          0.4     Iris-setosa   
..            ...          ...           ...          ...             ...   
144           6.7          3.0           5.2          2.3  Iris-virginica   
145           6.3          2.5           5.0          1.9  Iris-virginica   
146           6.5          3.0           5.2          2.0  Iris-virginica   
147           6.2          3.4           5.4          2.3  Iris-virginica   
148           5.9          3.0           5.1          1.8  Iris-virginica   

     class_type  
0             0  
1             0  
2             0  
3             0  
4             0  
..          ...  
144           2  
145           2  
146           2  
147           2  
148           2  

[149 rows x 6 columns]

In [9]:
data['class'].unique()

array(['Iris-setosa', 'Iris-versicolor', 'Iris-virginica'], dtype=object)

In [10]:
data=data.drop("class",axis=1)
data

sepal_length  sepal_width  petal_length  petal_width  class_type
0             4.9          3.0           1.4          0.2           0
1             4.7          3.2           1.3          0.2           0
2             4.6          3.1           1.5          0.2           0
3             5.0          3.6           1.4          0.2           0
4             5.4          3.9           1.7          0.4           0
..            ...          ...           ...          ...         ...
144           6.7          3.0           5.2          2.3           2
145           6.3          2.5           5.0          1.9           2
146           6.5          3.0           5.2          2.0           2
147           6.2          3.4           5.4          2.3           2
148           5.9          3.0           5.1          1.8           2

[149 rows x 5 columns]

Here, 1 is for Iris-sertosa <br>
       2 is for Iris-versicolor<br>
       3 if for Iris-virginica

In [11]:
x=data.iloc[:,:-1]
x

sepal_length  sepal_width  petal_length  petal_width
0             4.9          3.0           1.4          0.2
1             4.7          3.2           1.3          0.2
2             4.6          3.1           1.5          0.2
3             5.0          3.6           1.4          0.2
4             5.4          3.9           1.7          0.4
..            ...          ...           ...          ...
144           6.7          3.0           5.2          2.3
145           6.3          2.5           5.0          1.9
146           6.5          3.0           5.2          2.0
147           6.2          3.4           5.4          2.3
148           5.9          3.0           5.1          1.8

[149 rows x 4 columns]

In [12]:
y=data['class_type']
y

0      0
1      0
2      0
3      0
4      0
      ..
144    2
145    2
146    2
147    2
148    2
Name: class_type, Length: 149, dtype: int32

Splitting data for training and testing

In [13]:
from sklearn.model_selection import train_test_split
x_train,x_test,y_train,y_test=train_test_split(x,y,test_size=0.25,random_state=1)

In [14]:
x_train

sepal_length  sepal_width  petal_length  petal_width
17            5.7          3.8           1.7          0.3
113           5.8          2.8           5.1          2.4
143           6.7          3.3           5.7          2.5
69            5.9          3.2           4.8          1.8
75            6.8          2.8           4.8          1.4
..            ...          ...           ...          ...
133           6.1          2.6           5.6          1.4
137           6.0          3.0           4.8          1.8
72            6.1          2.8           4.7          1.2
140           6.9          3.1           5.1          2.3
37            4.4          3.0           1.3          0.2

[111 rows x 4 columns]

In [15]:
y_train

17     0
113    2
143    2
69     1
75     1
      ..
133    2
137    2
72     1
140    2
37     0
Name: class_type, Length: 111, dtype: int32

In [16]:
y_train

17     0
113    2
143    2
69     1
75     1
      ..
133    2
137    2
72     1
140    2
37     0
Name: class_type, Length: 111, dtype: int32

In [17]:
y_test

145    2
89     1
54     1
77     1
84     1
105    2
91     1
53     1
44     0
59     1
5      0
16     0
93     1
14     0
58     1
33     0
73     1
29     0
66     1
35     0
119    2
144    2
31     0
130    2
111    2
19     0
51     1
48     0
90     1
102    2
127    2
98     1
42     0
136    2
28     0
40     0
107    2
117    2
Name: class_type, dtype: int32

Importing Logistic Reression for evaluating the model

In [18]:
from sklearn.linear_model import LogisticRegression
model=LogisticRegression()

In [19]:
model.fit(x_train,y_train)

c:\users\rashmi\appdata\local\programs\python\python39\lib\site-packages\sklearn\linear_model\_logistic.py:763: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


LogisticRegression()

In [20]:
pred=model.predict(x_test)

In [21]:
from sklearn.metrics import confusion_matrix,accuracy_score

In [22]:
cm=confusion_matrix(pred,y_test)
cm

array([[13,  0,  0],
       [ 0, 14,  1],
       [ 0,  0, 10]], dtype=int64)

In [23]:
accuracy_score(pred,y_test)

0.9736842105263158

Accuracy of the model is high i.e. 0.97. It gives almost accurate prediction